subprojects/docs/src/docs/userguide/upgrading_version_4.adoc

```diff
@@ -328,6 +377,149 @@ To override this behavior, you can explicitly declare the configuration to use i
+
+[[rel5.0:default_tool_versions]]
+=== [5.0] New default versions for code quality plugins
+
+The default tool versions of the following code quality plugins have been updated:
+
+ * The http://checkstyle.sourceforge.net[Checkstyle] plugin now uses http://checkstyle.sourceforge.net/releasenotes.html#Release_8.12[8.12] instead of 6.19 by default.
+ * The http://codenarc.sourceforge.net[CodeNarc] plugin now uses https://github.com/CodeNarc/CodeNarc/blob/master/CHANGELOG.md#version-121-aug-2018[1.2.1] instead of 1.1 by default.
+ * The https://www.jacoco.org/jacoco/[JaCoCo] plugin now uses https://www.jacoco.org/jacoco/trunk/doc/changes.html[0.8.2] instead of 0.8.1 by default.
+ * The https://pmd.github.io/[PMD] plugin now uses https://pmd.github.io/pmd-6.8.0/pmd_release_notes.html#30-september-2018---680[6.8.0] instead of 5.6.1 by default.
++
+In addition, the default ruleset was changed from the now deprecated `java-basic` to `category/java/errorprone.xml`.
++
+We recommend configuring a ruleset explicitly, though.
+
+[[rel5.0:library_upgrades]]
+=== [5.0] Library upgrades
+Several libraries that are used by Gradle have been upgraded:
+
+ * Groovy was upgraded from 2.4.15 to http://groovy-lang.org/releasenotes/groovy-2.5.html[2.5.2].
+ * Ant has been upgraded from 1.9.11 to https://archive.apache.org/dist/ant/RELEASE-NOTES-1.9.13.html[1.9.13].
+ * The AWS SDK used to access S3 backed Maven/Ivy repositories has been upgraded from 1.11.267 to https://github.com/aws/aws-sdk-java/blob/master/CHANGELOG.md#111407-2018-09-11[1.11.407].
+ * The BND library used by the OSGi plugin has been upgraded from 3.4.0 to https://github.com/bndtools/bnd/wiki/Changes-in-4.0.0[4.0.0].
+ * The Google Cloud Storage JSON API Client Library used to access Google Cloud Storage backed Maven/Ivy repositories has been upgraded from v1-rev116-1.23.0 to v1-rev136-1.25.0.
+ * Ivy has been upgraded from 2.2.0 to http://ant.apache.org/ivy/history/2.3.0/release-notes.html[2.3.0].
+ * The JUnit Platform libraries used by the `Test` task have been upgraded from 1.0.3 to 1.3.1.
+ * The Maven Wagon libraries used to access Maven repositories have been upgraded from 2.4 to 3.0.0.
+ * SLF4J has been upgraded from 1.7.16 to https://www.slf4j.org/news.html[1.7.25].
+
+[[rel5.0:dependency_resolution]]
+=== [5.0] Fixes to dependency resolution
+
+Dependency resolutions fixes have been included in this release.
+By definition this could impact the set of resolved dependencies of your build.
+However the fixed issues are mostly about corner cases and combination with recent features and thus should have a limited impact.
+
+When a dependency constraint matched a real dependency, it was made part of the graph.
+However if for some reason the dependency was later evicted from the graph, the constraint remained present. Now, when the last non-constraint edge to a dependency disappears, all constraints for that dependency will be properly removed from the graph.
```

Comment:
What's this paragraph about? Is it introducing a specific fix? Or is this the only fix that the release made? We need to make a connection with the previous paragraph in some way.

It would also be useful to get a translation of this paragraph for "normal" users :smile: It's a bit hard to grok what it's saying.

Reply:
@gradle/dependency-management ? Can you explain it again for us?

Reply:
So that was from me 😉 

I think both paragraph should be merged / updated. What about:
> Various issues have been fixed in Gradle's dependency resolution for this release. By definition this could impact the set of resolved dependencies of your build.
> If you have been using the `IMPROVED_POM_SUPPORT` feature preview, playing with constraints or `prefer`, `reject` and other specific version indications, then make sure to take a good look at your dependency resolution results.

The original explanation is indeed hard to parse as the issue stems from multiple steps, having to happen in a specific order for the change in behavior to be observed. I think what I suggest above is easier and users should easily be able to say in which category they belong.